pkg/controller/machine/controller_test.go

```diff
@@ -1199,3 +1226,123 @@ func TestSetLifecycleHookConditions(t *testing.T) {
 		})
 	}
 }
+
+func TestSetPausedCondition(t *testing.T) {
+	t.Run("PausedCondition sets correctly", func(t *testing.T) {
+		g := NewWithT(t)
+		k8sClient, err := client.New(cfg, client.Options{})
+		g.Expect(err).ToNot(HaveOccurred())
+		act := newTestActuator()
+		r := &ReconcileMachine{
+			Client:   k8sClient,
+			scheme:   scheme.Scheme,
+			actuator: act,
+		}
+
+		// Set up the test namespace
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-paused-condition",
+			},
+		}
+		g.Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+		defer func() {
+			if err := k8sClient.Delete(ctx, namespace); err != nil {
+				t.Fatalf("error deleting namespace: %v", err)
+			}
+		}()
+
+		// Set up the test machine in a state where we expect the paused condition
+		// to be true
+		machine := &machinev1.Machine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "machine.openshift.io/v1beta1",
+				Kind:       "Machine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-machine",
+				Namespace:    namespace.Name,
+				Finalizers:   []string{machinev1.MachineFinalizer, metav1.FinalizerDeleteDependents},
+				Labels: map[string]string{
+					machinev1.MachineClusterIDLabel: "testcluster",
+				},
+			},
+			Spec: machinev1.MachineSpec{
+				AuthoritativeAPI: machinev1.MachineAuthorityClusterAPI,
+				ProviderSpec: machinev1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte("{}"),
+					},
+				},
+			},
+			Status: machinev1.MachineStatus{
+				AuthoritativeAPI: machinev1.MachineAuthorityClusterAPI,
+				Addresses: []corev1.NodeAddress{
+					{
+						Type:    corev1.NodeInternalIP,
+						Address: "0.0.0.0",
+					},
+				},
+			},
+		}
+
+		g.Expect(k8sClient.Create(ctx, machine)).To(Succeed())
+		defer func() {
+			if err := k8sClient.Delete(ctx, machine); err != nil {
+				t.Fatalf("error deleting machine: %v", err)
+			}
+		}()
+
+		machine.Status.AuthoritativeAPI = machinev1.MachineAuthorityClusterAPI
+		g.Expect(k8sClient.Status().Update(ctx, machine)).To(Succeed())
+
+		key := objectKey(machine)
+		g.Expect(k8sClient.Get(ctx, key, machine)).To(Succeed())
+		g.Expect(machine.Status.AuthoritativeAPI).To(Equal(machinev1.MachineAuthorityClusterAPI))
+
+		g.Eventually(func() bool {
+			_, err = r.Reconcile(ctx, reconcile.Request{NamespacedName: objectKey(machine)})
+			g.Expect(err).ToNot(HaveOccurred())
+
+			key := objectKey(machine)
+			if err := k8sClient.Get(ctx, key, machine); err != nil {
+				return false
+			}
+
+			// The condition is set to true
+			return conditions.IsTrue(machine, PausedCondition)
+		}, timeout).Should(BeTrue())
+
+		// Set the AuthoritativeAPI to Migrating as we must transition through it
+		machine.Status.AuthoritativeAPI = machinev1.MachineAuthorityMigrating
+		g.Expect(k8sClient.Status().Update(ctx, machine)).To(Succeed())
+
+		// Do we want to expect we remain paused during migration? Wouldn't this
+		// just instant pass?
```

Comment:
Yes it would, so you could use a `Consistently` to check that it doesn't get changed